lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -1389,10 +1389,9 @@ namespace Js
         return concatenated.ToString();
     }
 
-    Var RegexHelper::StringReplace(JavascriptString* match, JavascriptString* input, JavascriptFunction* replacefn)
+    Var RegexHelper::StringReplace(ScriptContext* scriptContext, JavascriptString* match, JavascriptString* input, JavascriptFunction* replacefn)
     {
         CharCount indexMatched = JavascriptString::strstr(input, match, true);
-        ScriptContext* scriptContext = replacefn->GetScriptContext();
         Assert(match->GetScriptContext() == scriptContext);
         Assert(input->GetScriptContext() == scriptContext);
```

Comment:
I meant is it required that both `math` and `input` should be in currentScriptContext?

---
In reply to: [103047018](https://github.com/Microsoft/ChakraCore/pull/2459#discussion_r103047018) [](ancestors = 103047018)